src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointGeometryModel3D.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Linq;
+using System.Runtime.InteropServices;
+using HelixToolkit.Wpf.SharpDX;
+using SharpDX;
+using SharpDX.Direct3D11;
+
+namespace Dynamo.Wpf.ViewModels.Watch3D
+{
+    /// <summary>
+    /// A struct describing the layout of a point vertex for Dynamo.
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential, Pack = 4)]
+    public struct DynamoPointVertex
+    {
+        public Vector4 Position;
+        public Color4 Color;
+        public Vector4 Parameters;
+        public const int SizeInBytes = 4 * (4 + 4 + 4);
+    }
+
+    /// <summary>
+    /// A Dynamo point class which supports the RenderCustom technique.
+    /// </summary>
+    public class DynamoPointGeometryModel3D : PointGeometryModel3D
+    {
+        public override int VertexSizeInBytes
+        {
+            get
+            {
+                return DynamoPointVertex.SizeInBytes;
+            }
+        }
+
+        /// <summary>
+        /// Override PointGeometryModel3D's Attach method to 
+        /// provide a buffer of DynamoPointVertices
+        /// </summary>
+        public override void Attach(IRenderHost host)
+        {
+            var techManager = host.RenderTechniquesManager;
+
+            renderTechnique = techManager.RenderTechniques[DefaultRenderTechniqueNames.Points];
+            base.Attach(host);
+
+            if (Geometry == null)
+                return;
+
+            if (host.SupportDeferredRender)
+            {
+                if (renderHost.RenderTechnique == techManager.RenderTechniques[DeferredRenderTechniqueNames.Deferred] || 
+                    renderHost.RenderTechnique == techManager.RenderTechniques[DeferredRenderTechniqueNames.GBuffer])
+                    return;
+            }
+
+            vertexLayout = host.EffectsManager.GetLayout(renderTechnique);
+            effectTechnique = effect.GetTechniqueByName(renderTechnique.Name);
+
+            effectTransforms = new EffectTransformVariables(effect);
+
+            var geometry = Geometry as PointGeometry3D;
+            if (geometry != null)
+            {          
+                vertexBuffer = Device.CreateBuffer(BindFlags.VertexBuffer, VertexSizeInBytes, CreateVertexArray());
+            }
+
+            vViewport = effect.GetVariableByName("vViewport").AsVector();
+            //this.vFrustum = effect.GetVariableByName("vFrustum").AsVector();
```

Comment:
You know what I'm going to say about commented out code. I'd assume you have a reason to keep it - if so, document accordingly.